cache/class-vary-cache.php

```diff
@@ -90,19 +98,26 @@ public static function clear_groups() {
 	 *
 	 * @param  string $group  Group name to vary the request on.
 	 * @param  string $value A value for the group.
+	 * @return WP_Error|boolean
 	 */
 	public static function set_group_for_user( $group, $value ) {
 		// TODO: make sure headers aren't already sent
 		// TODO: only send header if we added or changed things
 		// TODO: don't set the cookie if was already set on the request
-		// validate, process $group, etc.
+		if ( strpos( $group, self::GROUP_SEPARATOR ) !== false || strpos( $group, self::VALUE_SEPARATOR ) !== false ) {
+			return new WP_Error( 'invalid_vary_group_name', sprintf( 'Failed to register group; cannot use the delimiter values (`%s` or `%s`) in the group name', self::GROUP_SEPARATOR, self::VALUE_SEPARATOR ) );
```

Comment:
I think it's worth still triggering a warning via `trigger_error` as that's a good signal that something is wrong.

Reply:
Actually, looking at anticipated usage, `register_groups` is more of a fire-and-forget function (there's not much happening) so `trigger_error` is probably better there.

Whereas `set_group_for_user` should probably require handling the result (and just the WP_Error should be good enough).